mm2src/mm2_bitcoin/keys/src/address.rs

```diff
@@ -170,7 +170,14 @@ impl DisplayLayout for Address {
 }
 
 impl fmt::Display for Address {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result { self.layout().to_base58().fmt(f) }
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match &self.addr_format {
+            AddressFormat::Segwit => SegwitAddress::new(&self.hash, self.hrp.clone().unwrap_or_default())
```

Comment:
`unwrap_or_default` is questionable here: it can display the invalid address when `hrp` is none. I think we can consider missing hrp as developer error here and replace the `unwrap_or_default` with expect.